### PR TITLE
lower utp fragment size by 512 bytes

### PR DESCRIPTION
**Pull request by majestrate**
*  lower utp fragment size by 512 bytes
* backport crash fix from master

### CODE REVIEW

**Comment by majestrate** [comment]
additionally forwardport today's fixes from master

File: docs/old-wire-protocol.txt [hunk 1]
```diff
@@ -30,13 +30,13 @@ B.rx_K = B_K
 
 the initial value of a_K is HS(A.k) and b_K is HS(B.k)
 
-1120 byte fragments are sent over UTP in an ordered fashion.
+608 byte fragments are sent over UTP in an ordered fashion.
```
Comment (michael-loki) [inline docs/old-wire-protocol.txt]: 👍 for updating the protocol.

File: llarp/service/endpoint.cpp [hunk 2]
```diff
@@ -254,6 +260,17 @@ namespace llarp
...
+      // deregister dead sessions
+      {
+        auto itr = m_DeadSessions.begin();
+        while(itr != m_DeadSessions.end())
+        {
+          if(itr->second->IsDone(now) && itr->second->ShouldRemove())
+            itr = m_DeadSessions.erase(itr);
+          else
+            ++itr;
+        }
```
Comment (michael-loki) [inline llarp/service/endpoint.cpp]: you could replace this block with:

```c++
std::erase(std::remove_if(m_DeadSessions.begin(), m_DeadSessions.end(), [now](const auto& val) {
  return (val.second->IsDone(now) && val.second->ShouldRemove());
}), m_DeadSessions.end());
```

Comment (majestrate) [reply llarp/service/endpoint.cpp]: yea i could, should i?

Comment (michael-loki) [reply llarp/service/endpoint.cpp]: up to you. I prefer the STL version, but I'm a C++ weenie.

**Pull request merged**